runtime/runtime-core/native/src/aws/smithy/kotlin/runtime/time/InstantNative.kt

```diff
@@ -5,92 +5,118 @@
+
+    actual override fun compareTo(other: Instant): Int = delegate.compareTo(other.delegate)
 
-    public actual val epochSeconds: Long
-        get() = TODO("Not yet implemented")
-    public actual val nanosecondsOfSecond: Int
-        get() = TODO("Not yet implemented")
+    public actual val epochSeconds: Long = delegate.epochSeconds
+    public actual val nanosecondsOfSecond: Int = delegate.nanosecondsOfSecond
 
     /**
      * Encode the [Instant] as a string into the format specified by [TimestampFormat]
      */
-    public actual fun format(fmt: TimestampFormat): String {
-        TODO("Not yet implemented")
+    public actual fun format(fmt: TimestampFormat): String = when (fmt) {
+        TimestampFormat.ISO_8601 -> delegate.truncateToMicros().format(DateTimeFormats.ISO_8601)
+        TimestampFormat.EPOCH_SECONDS -> {
+            val s = delegate.epochSeconds.toString()
+            val ns = if (delegate.nanosecondsOfSecond != 0) {
+                ".${delegate.nanosecondsOfSecond.toString().padStart(9, '0').trimEnd('0')}"
+            } else {
+                ""
+            }
+            s + ns
+        }
+        else -> delegate.format(fmt.asDateTimeFormat())
     }
 
     /**
      * Returns an instant that is the result of adding the specified [duration] to this instant.
      * NOTE: Duration may be negative in which case the returned Instant will be earlier than this Instant.
      */
-    public actual operator fun plus(duration: Duration): Instant {
-        TODO("Not yet implemented")
-    }
+    public actual operator fun plus(duration: Duration): Instant = Instant(delegate + duration)
 
     /**
      * Returns an instant that is the result of subtracting the specified [duration] from this instant.
      * NOTE: Duration may be negative in which case the returned Instant will be later than this Instant.
      */
-    public actual operator fun minus(duration: Duration): Instant {
-        TODO("Not yet implemented")
-    }
+    public actual operator fun minus(duration: Duration): Instant = Instant(delegate - duration)
 
-    public actual operator fun minus(other: Instant): Duration {
-        TODO("Not yet implemented")
-    }
+    public actual operator fun minus(other: Instant): Duration = delegate - other.delegate
 
     public actual companion object {
         /**
          * Parse an ISO-8601 formatted string into an [Instant]
          */
         public actual fun fromIso8601(ts: String): Instant {
-            TODO("Not yet implemented")
+            var parsed = DateTimeFormats.ISO_8601.parse(ts)
+
+            // Handle leap seconds (23:59:59)
+            parsed.second = parsed.second?.coerceAtMost(59)
+
+            // Handle leap hours (24:00:00)
+            return if (parsed.hour == 24) {
+                parsed.hour = 0
+                Instant(parsed.toInstantUsingOffset() + 1.days)
+            } else {
+                Instant(parsed.toInstantUsingOffset())
+            }
```

Comment:
**Correctness**: What's a leap hour?

Reply:
Sorry not a leap hour, it's a midnight timestamp

https://github.com/smithy-lang/smithy-kotlin/blob/kn-time/runtime/runtime-core/common/test/aws/smithy/kotlin/runtime/time/InstantTest.kt#L59-L60